universal-application-tool-0.0.1/test/services/applicant/ApplicantQuestionTest.java

```diff
@@ -66,7 +66,30 @@ public void textQuestion_withPresentApplicantData() {
         new ApplicantQuestion(textQuestionDefinition, applicantData);
     ApplicantQuestion.TextQuestion textQuestion = applicantQuestion.getTextQuestion();
 
-    assertThat(textQuestion.hasErrors()).isFalse();
+    assertThat(textQuestion.hasTypeSpecificErrors()).isFalse();
+    assertThat(textQuestion.getTextValue().get()).isEqualTo("hello");
+  }
+
+  @Test
+  public void textQuestion_withPresentApplicantData_failsValidation() {
+    TextQuestionDefinition question =
```

Comment:
_Nit(non-blocking):_ I feel like`QuestionDefinitionBuilder` makes this more readable.

```
TextQuestionDefinition question = new QuestionDefinitionBuilder()
    .setId(1L)
    .setQuestionType(QuestionType.TEXT)
    .setName("question name")
    .setPath("applicant.my.path.name")
    .setDescription("description")
    .setQuestionText(ImmutableMap.of(Locale.ENGLISH, "question?"))
    .setQuestionHelpText(ImmutableMap.of(Locale.ENGLISH, "help text"))
    .build();
```

Reply:
This has the slightly undesirable side effect where `build()` returns a `QuestionDefinition` not `TextQuestionDefinition` so we have to down cast it. Not a big deal as this is in test.

Reply:
Done.

Reply:
Ahh interesting. Good to consider.